### PR TITLE
[CI] Use Ubuntu 24.04 for dev-igc and unstable

### PR DESCRIPTION
**Pull request by sarnex**
dev_igc and unstable need Ubuntu 24.04 now based on the IGC packaging, so move our Dockerfiles and CI to use 24.04 for these.

I needed to manually bump the dev-igc version because the artifacts for the current one are expired so they can't be downloaded anymore so we can't make the Docker image.

The dev-igc CI run is expected to fail here, it always uses the official intel/llvm Docker images, even if the current PR updates them, so we need to wait for this PR to be merged to test it. I'll watch it once we merge.

### CODE REVIEW

File: devops/dependencies-igc-dev.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "linux": {
     "igc_dev": {
-      "github_tag": "igc-dev-6ee988a",
-      "version": "6ee988a",
-      "updated_at": "2024-11-26T15:44:10Z",
-      "url": "https://api.github.com/repos/intel/intel-graphics-compiler/actions/artifacts/2239640503/zip",
+      "github_tag": "igc-dev-3db59df",
+      "version": "3db59df ",
```
Comment (YuriPlyakhin) [inline devops/dependencies-igc-dev.json:5]: There is one space at the end of value. Will it cause problems?

**Review by YuriPlyakhin — APPROVED** [review]
one potential problem, otherwise LGTM.

File: devops/dependencies-igc-dev.json (hunk 1 above)
Comment (sarnex) [reply devops/dependencies-igc-dev.json:5]: wow nice catch, will fix, thanks!

**Review by cperkinsintel — APPROVED** [review]

**Comment by sarnex** [comment]
@bader Ping for doc review? Thanks

File: sycl/doc/developer/DockerBKMs.md [hunk 2]
```diff
@@ -41,17 +41,20 @@ The following containers are publicly available for DPC++ compiler development:
...
+   Ubuntu 22.04 base container + pre-installed Intel drivers.
+   The image comes in two flavors/tags:
    * `latest`: Intel drivers are downloaded from release/tag and saved in
     dependencies.json. The drivers are tested/validated everytime we upgrade
     the driver.
+   * `alldeps`: Includes the same Intel drivers as `latest`, as well as the
+   development kits for NVidia/AMD from the `ubuntu2204_build` container.
+- `ghcr.io/intel/llvm/ubuntu2404_intel_drivers`: contains everything from the
+   Ubuntu 24.04 base container + pre-installed Intel drivers.
+   The image comes in two flavors/tags:
```
Comment (bader) [inline sycl/doc/developer/DockerBKMs.md:53]: I believe we should provide `latest` and `alldeps` for Ubuntu 24.04 as well. Any plans for enabling them in the future?

**Review by bader — APPROVED** [review]

File: sycl/doc/developer/DockerBKMs.md (hunk 2 above)
Comment (sarnex) [reply sycl/doc/developer/DockerBKMs.md:53]: sure, ill do it after i fix the igc-dev builds which is my overall goal with these PRs

**Pull request merged**